kpops/component_handlers/schema_handler/schema_handler.py

```diff
@@ -19,28 +20,24 @@
 
 
 class SchemaHandler:
-    _schema_provider: SchemaProvider | None = None
-
     def __init__(self, url: str, components_module: str | None):
         self.schema_registry_client = SchemaRegistryClient(url)
         self.components_module = components_module
 
-    @property
+    @cached_property
     def schema_provider(self) -> SchemaProvider:
-        if not self._schema_provider:
-            try:
-                if not self.components_module:
-                    raise ValueError(
-                        "The Schema Registry URL is set but you haven't specified the component module path. Please provide a valid component module path where your SchemaProvider implementation exists."
-                    )
-                schema_provider_class = find_class(self.components_module, SchemaProvider)  # type: ignore[type-abstract]
-                self._schema_provider = schema_provider_class()
-            except ClassNotFoundError:
+        try:
+            if not self.components_module:
                 raise ValueError(
-                    f"No schema provider found in components module {self.components_module}. "
-                    f"Either implement it by inheriting from {SchemaProvider.__module__}.{SchemaProvider.__name__} and implementing its abstract method."
+                    "The Schema Registry URL is set but you haven't specified the component module path. Please provide a valid component module path where your SchemaProvider implementation exists."
                 )
-        return self._schema_provider
+            schema_provider_class = find_class(self.components_module, SchemaProvider)  # type: ignore[type-abstract]
+            return schema_provider_class()
+        except ClassNotFoundError:
+            raise ValueError(
+                f"No schema provider found in components module {self.components_module}. "
+                f"Either implement it by inheriting from {SchemaProvider.__module__}.{SchemaProvider.__name__} and implementing its abstract method."
```

Comment:
The "Either" probably lost its meaning here from a previous refactoring. There only seems to be one solution